### PR TITLE
chore: removes eslint warnings for enums simplificiation

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
Removes complexity of enum simplification.

```
/Users/lagoni/Documents/AsyncAPI/generator-model-sdk/src/simplification/SimplifyEnums.ts
  10:25  warning  Refactor this function to reduce its Cognitive Complexity from 28 to the 15 allowed  sonarjs/cognitive-complexity
```

**Related issue(s)**
part of #53

### CODE REVIEW

File: src/simplification/SimplifyEnums.ts [hunk 1]
```diff
@@ -9,56 +9,84 @@ type Output = any[] | undefined;
...
+}
+
+/**
+ * Ensure enums in not are never included.
+ * 
+ * @param schema currently searching in
+ * @param existingEnums which have already been found
+ * @param seenSchemas already seen schemas and their respectable output
+ */
+function ensureNotEnumsAreRemoved(schema: Schema, existingEnums: any[], seenSchemas: Map<any, Output>) {
```
Comment (magicmatatjahu) [inline src/simplification/SimplifyEnums.ts:51]: `ensureNotEnums` is enought :)

**Review by magicmatatjahu — APPROVED** [review]

File: src/simplification/SimplifyEnums.ts [hunk 2]
```diff
@@ -9,56 +9,84 @@ type Output = any[] | undefined;
  */
 export default function simplifyEnums(schema: Schema | boolean, seenSchemas: Map<any, Output> = new Map()): Output {
   if (typeof schema !== 'boolean') {
```
Comment (magicmatatjahu) [inline src/simplification/SimplifyEnums.ts:11]: Also I think that we should go with this approach with handling boolean schemas:

```ts
if (typeof schema === 'boolean') {
  return udnefined;
}

// ...rest of logic
```

For me this is better to read and understand. Of course I can do that in follow up PR after yours.

Comment (jonaslagoni) [reply src/simplification/SimplifyEnums.ts:11]: If we want consistency I think we should do that in another PR yes 😄

File: src/simplification/SimplifyEnums.ts (hunk 1 above)
Comment (jonaslagoni) [reply src/simplification/SimplifyEnums.ts:51]: True 😄

**Pull request merged**